partner_merge_group/wizard/partner_merge_views.py

```diff
@@ -0,0 +1,28 @@
+# -*- coding: utf-8 -*-
+# Copyright 2019 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, models, _
+
+
+class MergePartnerAutomatic(models.TransientModel):
+    _inherit = 'base.partner.merge.automatic.wizard'
+
+    def _merge(self, partner_ids, dst_partner=None):
+        super(MergePartnerAutomatic, self).sudo(self.user.id).with_context(
```

Comment:
You have an infinite recursive loop because `sudo()` must be done on self, not on the whole `super()` statement.
Also, always better to use `return` in case `_merge` returns something.